oauth-token-client/src/main/kotlin/com/gxf/utilities/spring/oauth/config/OAuthClientProperties.kt

```diff
@@ -5,13 +5,14 @@ SPDX-License-Identifier: Apache-2.0
 */
 package com.gxf.utilities.spring.oauth.config
 
+import com.gxf.utilities.spring.oauth.config.condition.OAuthEnabledCondition
 import org.springframework.beans.factory.annotation.Value
 import org.springframework.context.annotation.Conditional
 import org.springframework.context.annotation.Configuration
 import org.springframework.core.io.Resource
 
 @Configuration
-@Conditional(OauthClientCondition::class)
+@Conditional(OAuthEnabledCondition::class)
 class OAuthClientProperties(
     @Value("\${oauth.client.client-id}")
```

Comment:
Als je hier een property hebt, gekoppeld (noodzakelijk) voor OAuth, dan kun je ook een andere conditionaliteit gaan gebruiken:
```java
@ConditionalOnProperty(
  value="logging.enabled", 
  havingValue = "true", 
  matchIfMissing = true)
```
En misschien een `@ConditionalOnMissingBean` op de `NoTokenProvider` (ik weet niet of dat werkt op een class met interface)

Reply:
Niet handig, omdat we dit ook in niet-Spring-Boot omgevingen gaan gebruiken